src/wrappers.jl

```diff
@@ -142,20 +161,20 @@ function SciMLBase.solve(cache::LinearCache, alg::KrylovJL; kwargs...)
               alg.kwargs...)
 
     if cache.cacheval isa Krylov.CgSolver
-        alg.Pr != LinearAlgebra.I  &&
+        N != LinearAlgebra.I  &&
             @warn "$(alg.KrylovAlg) doesn't support right preconditioning."
```

Comment:
Symmetric Krylov methods, such as CG or MINRES are only using split preconditioning.
Instead of solving `Ax = b`, you solve `M^{1/2} A M^{1/2} x = M^{1/2} b` because you need to keep a symmetric linear system. `M` must be is SPD matrix for that reason.